datajs/src/file-base.js

```diff
@@ -145,45 +145,12 @@ export class FileInterface extends File {
   }
 
 
-
-
-  /**
-   *
-   * @param {string} hashType - Should be md5 or sha256 
-   * @param {string} cbProgress - Should be a callback to track the progress
-   */
-  async generateHash(hashType, cbProgress) {
-    return new Promise((resolve, reject) => {
-      let newHash =  hashType === "md5" ? crypto.createHash('md5') : crypto.createHash('sha256');
-      readChunk(this.descriptor, (chunk, offs, total) => {
-        newHash.update(chunk);
-        if (cbProgress) {
-          cbProgress(offs / total);
-        }
-      }, err => {
-        if (err) {
-          reject(err);
-        } else {
-          let hashHex = newHash.digest('hex')
-          resolve(hashHex);
-        }
-      });
-    });
-  }
-
-  /**
-   *
-   * @param {string} cbProgress - Should be a callback to track the progress
-   */
-  async hash(cbProgress) {
-    return this.generateHash("md5", cbProgress)
-  }
-
-  /**
-   *
-   * @param {string} cbProgress - Should be a callback to track the progress
+   /**
+   * Calculates the hash of a file
+   * @param {string} hashType - md5/sha256 type of hash algorithm to use
    */
-   async hashSha256(cbProgress) {
-    return this.generateHash("sha256", cbProgress)
+  async hash(hashType='sha256') {
+    let stream = webToNodeStream(this.descriptor.stream())
```

Comment:
Call it's `stream()` function instead of this. Move `webToNodeStream()` to `sream()` class method